source/slang/slang-check-decl.cpp

```diff
@@ -12945,6 +12945,27 @@ CapabilitySet getStatementCapabilityUsage(SemanticsVisitor* visitor, Stmt* stmt)
 
 void SemanticsDeclCapabilityVisitor::checkVarDeclCommon(VarDeclBase* varDecl)
 {
+    auto declaredCaps = getDeclaredCapabilitySet(varDecl);
```

Comment:
This check should only be needed for global variables (variables that doesn't have a function parent), since for local variables their requirements are already propagated to the enclosing function and we are already diagnosing errors on the function.